src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish

```diff
@@ -0,0 +1,91 @@
+# ----------------------------------------------------------------------------
+#   Visual Studio Code terminal integration for fish
+#   Copyright (c) Microsoft Corporation. All rights reserved.
+#   Licensed under the MIT License. See License.txt in the project root for license information.
+# ----------------------------------------------------------------------------
+# Manual installation:
+#
+#   (1) Add the following to the end of `$__fish_config_dir/config.fish`:
+#
+#         string match -q "$TERM_PROGRAM" "vscode"
+#         and . (code --locate-shell-integration-path fish)
+#
+#   (2) Restart fish.
+# ----------------------------------------------------------------------------
+# TODO: Confirm all escape sequences once they are finalized.
+# See microsoft/vscode#155639 and microsoft/vscode#139400 for discussion.
+# ----------------------------------------------------------------------------
+
+# Don't run in scripts, other terminals, or more than once per session.
+status is-interactive
+and string match --quiet "$TERM_PROGRAM" "vscode"
+and ! set --query VSCODE_SHELL_INTEGRATION
+or return
```

Comment:
When run in a fish terminal on Insiders I see this:

```
~/dev/Microsoft/vscode/src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish (line 23): 'return' outside of function definition
or return
   ^
from sourcing file ~/dev/Microsoft/vscode/src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish
        called on line 359 of file /usr/local/share/fish/config.fish

in function '.'
        called on line 3 of file ~/.config/fish/config.fish

from sourcing file ~/.config/fish/config.fish
        called during startup

source: Error while reading file '/Users/daimms/dev/Microsoft/vscode/src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish'
Welcome to fish, the friendly interactive shell
daimms@Daniels-MBP ~/d/M/vscode> 
```

Should this be `or exit`?

Reply:
I can't reproduce this error, unfortunately, but it probably should have been `exit` in the first place anyway, so I changed it. Please let me know if you still run into problems.

Reply:
The behavior of `return` changed in fish [3.4.0](https://fishshell.com/docs/current/relnotes.html#fish-3-4-0-released-march-12-2022):

> `return` can now be used outside functions. In scripts, it does the same thing as `exit`. In interactive mode,it sets $status without exiting

Reply:
Good to know, that was before I updated fish 👍